File: LinkedLists/LinkedList.java
import java.util.Scanner;

class Node {
    int data;
    Node next;

    Node(int data){
        this.data = data;
        next = null;
    }
}

public class LinkedList {

    static Node head = null;

    static void insertBegin(int val){
        Node newNode = new Node(val);
        newNode.next = head;
        head = newNode;
    }

    static void insertEnd(int val){
        Node newNode = new Node(val);

        if(head==null){
            head=newNode;
            return;
        }

        Node temp=head;

        while(temp.next!=null)
            temp=temp.next;

        temp.next=newNode;
    }

    static void insertPos(int val,int pos){

        Node newNode=new Node(val);
        Node temp=head;

        for(int i=1;i<pos-1;i++)
            temp=temp.next;

        newNode.next=temp.next;
        temp.next=newNode;
    }

    static void deleteBegin(){
        if(head==null){
            System.out.println("List is empty!");
            return;
        }
        head=head.next;
    }

    static void deleteEnd(){

        if(head==null){
            System.out.println("List is empty!");
            return;
        }

        Node temp=head,prev=null;

        while(temp.next!=null){
            prev=temp;
            temp=temp.next;
        }

        prev.next=null;
    }

    static void deletePos(int pos){

        if(head==null){
            System.out.println("List is empty!");
            return;
        }

        Node temp=head,prev=null;

        for(int i=1;i<pos;i++){
            prev=temp;
            temp=temp.next;
        }

        prev.next=temp.next;
    }

    static void display(){

        Node temp=head;

        while(temp!=null){
            System.out.print(temp.data+" -> ");
            temp=temp.next;
        }
        System.out.println("null");
    }
    public static void main(String[] args){

        Scanner sc=new Scanner(System.in);

        while(true){

            System.out.println("1.Insert Begin");
            System.out.println("2.Insert End");
            System.out.println("3.Insert Position");
            System.out.println("4.Delete Begin");
            System.out.println("5.Delete End");
            System.out.println("6.Delete Position");
            System.out.println("7.Display");
            System.out.println("8.Exit");

            int ch=sc.nextInt();

            switch(ch){

                case 1:
                    System.out.println("Enter value to insert at beginning: ");
                    insertBegin(sc.nextInt());
                    break;

                case 2:
                    System.out.println("Enter value to insert at end: ");
                    insertEnd(sc.nextInt());
                    break;

                case 3:
                    System.out.println("Enter value and position to insert: ");
                    insertPos(sc.nextInt(),sc.nextInt());
                    break;

                case 4:
                    System.out.println("Deleting from beginning...");
                    deleteBegin();
                    break;

                case 5:
                    System.out.println("Deleting from end...");
                    deleteEnd();
                    break;

                case 6:
                    System.out.println("Enter position to delete: ");
                    deletePos(sc.nextInt());
                    break;

                case 7:
                    System.out.println("Displaying list...");
                    display();
                    break;

                case 8:
                    sc.close();
                    System.exit(0);
            }
        }
    }
}

File: LinkedLists/LinkedList.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node* next;
};

struct Node* head = NULL;

void insertBegin(int val){
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = val;
    newNode->next = head;
    head = newNode;
}

void insertEnd(int val){
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
    struct Node* temp = head;

    newNode->data = val;
    newNode->next = NULL;

    if(head == NULL){
        head = newNode;
        return;
    }

    while(temp->next != NULL)
        temp = temp->next;

    temp->next = newNode;
}

void insertPos(int val, int pos){
    struct Node *newNode, *temp=head;
    int i;

    newNode = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = val;

    for(i=1;i<pos-1;i++)
        temp=temp->next;

    newNode->next=temp->next;
    temp->next=newNode;
}

void deleteBegin(){
    struct Node* temp=head;

    if(head==NULL) return;

    head=head->next;
    free(temp);
}

void deleteEnd(){
    struct Node *temp=head,*prev;

    if(head==NULL) return;

    if(head->next==NULL){
        free(head);
        head=NULL;
        return;
    }

    while(temp->next!=NULL){
        prev=temp;
        temp=temp->next;
    }

    prev->next=NULL;
    free(temp);
}

void deletePos(int pos){
    struct Node *temp=head,*prev;
    int i;

    for(i=1;i<pos;i++){
        prev=temp;
        temp=temp->next;
    }

    prev->next=temp->next;
    free(temp);
}

void display(){
    struct Node* temp=head;

    while(temp!=NULL){
        printf("%d -> ",temp->data);
        temp=temp->next;
    }
    printf("NULL\n");
}

int main(){
    int ch,val,pos;

    while(1){
        printf("\n1.Insert Begin\n2.Insert End\n3.Insert Position\n");
        printf("4.Delete Begin\n5.Delete End\n6.Delete Position\n");
        printf("7.Display\n8.Exit\n");

        scanf("%d",&ch);

        switch(ch){

            case 1:
                printf("Enter value to insert at beginning: ");
                scanf("%d",&val);
                insertBegin(val);
                break;

            case 2:
                printf("Enter value to insert at end: ");
                scanf("%d",&val);
                insertEnd(val);
                break;

            case 3:
                printf("Enter value and position to insert: ");
                scanf("%d%d",&val,&pos);
                insertPos(val,pos);
                break;

            case 4:
                printf("Deleting from beginning...\n");
                deleteBegin();
                break;

            case 5:
                printf("Deleting from end...\n");
                deleteEnd();
                break;

            case 6:
                printf("Enter position to delete: ");
                scanf("%d",&pos);
                deletePos(pos);
                break;

            case 7:
                printf("Displaying list...\n");
                display();
                break;

            case 8:
                exit(0);
        }
    }
}